pre_processing/set_ecmwf.F90

```diff
@@ -94,6 +98,10 @@ subroutine set_ecmwf(granule, opts, nwp_flag, imager_geolocation, &
    character     :: cera_year*4, cera_month*2, cera_day*2, cera_hour*2
    character     :: cera_year2*4, cera_month2*2, cera_day2*2, cera_hour2*2
 
+   ! If we're reading BADC ERA5 data, we default to one file every hour,
+   ! rather than 6-hours 
+   if (nwp_flag .eq. 2) opts%nwp_time_factor = 1.0
```

Comment:
I don't think this is the right place for this code. This factor is set by the driver file, and no one would expect that to be overridden in some other routine. Either,

- This should be removed from the driver file and set_ecmwf should contain a case statement that translates `nwp_flag` into an appropriate flag. I assume we didn't do that in the first place because the NWP files might have any spacing if they're downloaded by the user rather than in a repository.
- Your change should be [near here](https://github.com/ORAC-CC/orac/blob/cfe81c518c70f2f552dc420854309d5c22cfc0c6/pre_processing/orac_preproc.F90#L483) so the override happens around where the driver file was read.

Reply:
Fair enough. I've moved it to `orac_preproc.F90`, as suggested (i.e. your second option).